### PR TITLE
fix log rotate issue 

### PR DESCRIPTION
**Pull request by KjellKod**
Setting path to directory that doesn't end with a trailing '/' caused an issue where the first log got compressed but then the second log file change the name so it got concatenated with the directory name 
i.e. 
```
directory "hello", fileneme: "mylog" --> hello/mylog.log  at first run
                          compressed log    --> hellomylog.log after compression
```

Now it will be fine regardless if you have trailing slash or not with the directory name. 
```
directory "hello", fileneme: "mylog" --> hello/mylog.log  at first run
                          compressed log    --> hello/mylog.log after compression
```

### CODE REVIEW

File: logrotate/CMakeLists.txt [hunk 1]
```diff
@@ -82,6 +82,7 @@ SET(${LIBRARY_TO_BUILD}_VERSION_STRING)
 SET_TARGET_PROPERTIES(${LIBRARY_TO_BUILD} PROPERTIES LINKER_LANGUAGE CXX)
 TARGET_LINK_LIBRARIES(${LIBRARY_TO_BUILD} ${TCMALLOC} ${Boost_LIBRARIES} ${ZLIB_LIBRARIES} ${G3LOG})
 
+MESSAGE(".... g3log: ${G3LOG}")
```
Comment (KjellKod) [inline logrotate/CMakeLists.txt]: I suggest instead to print the location of all libraries

Comment (KjellKod) [inline logrotate/CMakeLists.txt]: remove tcmalloc

File: logrotate/src/LogRotate.cpp [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 #include "g3sinks/LogRotate.h"
 #include "LogRotateHelper.ipp"
+#include <iostream>
```
Comment (KjellKod) [inline logrotate/src/LogRotate.cpp]: remove

File: logrotate/src/LogRotateHelper.ipp [hunk 3]
```diff
@@ -96,12 +96,9 @@ LogRotateHelper::LogRotateHelper(const std::string& log_prefix, const std::strin
       abort();
    }
 
-    std::string file_name = createLogFileName(log_prefix_backup_);
-    log_file_with_path_ = pathSanityFix(log_directory, file_name);
-    outptr_ = createLogFile(log_file_with_path_);
+    auto logfile = changeLogFile(log_directory, log_prefix_backup_);
```
Comment (KjellKod) [inline logrotate/src/LogRotateHelper.ipp]: remove all couts

File: logrotate/CMakeLists.txt [hunk 4]
```diff
@@ -80,8 +80,11 @@ ENDIF()
 
 SET(${LIBRARY_TO_BUILD}_VERSION_STRING)
 SET_TARGET_PROPERTIES(${LIBRARY_TO_BUILD} PROPERTIES LINKER_LANGUAGE CXX)
-TARGET_LINK_LIBRARIES(${LIBRARY_TO_BUILD} ${TCMALLOC} ${Boost_LIBRARIES} ${ZLIB_LIBRARIES} ${G3LOG})
+TARGET_LINK_LIBRARIES(${LIBRARY_TO_BUILD}  ${Boost_LIBRARIES} ${ZLIB_LIBRARIES} ${G3LOG})
```
Comment (shwetabhandare) [inline logrotate/CMakeLists.txt]: Do we still have boost presence?

File: logrotate/src/LogRotate.cpp (hunk 2 above)
Comment (shwetabhandare) [inline logrotate/src/LogRotate.cpp]: why do you need iostream?

File: logrotate/src/LogRotateHelper.ipp [hunk 5]
```diff
@@ -168,20 +164,22 @@ std::string LogRotateHelper::changeLogFile(const std::string& directory, const s
       file_name = log_prefix_backup_;
    }
 
-   std::string prospect_log = createLogFileName(directory + file_name);
```
Comment (shwetabhandare) [inline logrotate/src/LogRotateHelper.ipp]: I think you can keep the createLogFileName but have both these functions inside that main function. that way you get the log file to create in the next line.

File: logrotate/test/FilterTest.cpp [hunk 6]
```diff
@@ -66,8 +66,8 @@ TEST_F(FilterTest, CreateObjectUsingHelper) {
     check = access(name.c_str(), F_OK); // check that the file exists
     EXPECT_EQ(check, 0) << std::strerror(errno) << " : " << name;
     auto content = ReadContent(name);
-    EXPECT_TRUE(Exists(content, "g3log: created log file at:"));
-    EXPECT_TRUE(Exists(content, "file shutdown at:"));
+    EXPECT_TRUE(Exists(content, "g3log: created log file at:")) << ", content: [" << "]" << ", name: " << name;
+    EXPECT_TRUE(Exists(content, "file shutdown at:")) << ", ontent: [" << "]"<< ", name: " << name;
```
Comment (d-led) [inline logrotate/test/FilterTest.cpp]: little typo: "ontent"?

File: logrotate/test/FilterTest.cpp [hunk 7]
```diff
@@ -56,7 +56,7 @@ TEST_F(FilterTest, CreateObject) {
 
 
 TEST_F(FilterTest, CreateObjectUsingHelper) {
-    auto name = std::string{_directory + _filename + ".log"};
+    auto name = std::string{_directory +"/" + _filename + ".log"};
```
Comment (d-led) [inline logrotate/test/FilterTest.cpp]: ok

File: logrotate/CMakeLists.txt (hunk 4 above)
Comment (KjellKod) [inline logrotate/CMakeLists.txt]: Yes.

I think it can be removed but I haven't had time to look into it

File: logrotate/src/LogRotate.cpp (hunk 2 above)
Comment (KjellKod) [inline logrotate/src/LogRotate.cpp]: cerr

File: logrotate/src/LogRotateHelper.ipp (hunk 5 above)
Comment (KjellKod) [inline logrotate/src/LogRotateHelper.ipp]: I don't follow your line of reasoning.

File: logrotate/test/FilterTest.cpp (hunk 6 above)
Comment (KjellKod) [inline logrotate/test/FilterTest.cpp]: Yup. Thanks

File: logrotate/src/LogRotateHelper.ipp (hunk 5 above)
Comment (KjellKod) [inline logrotate/src/LogRotateHelper.ipp]: The only real change in this function is adding the 'createPath', previously called `pathSanityFix` (_the old name was bad_).  Now having the constructor also calling this function removes some duplication.

File: logrotate/CMakeLists.txt (hunk 4 above)
Comment (shwetabhandare) [inline logrotate/CMakeLists.txt]: I was just surprised you had this in your repo! No offence.

**Comment by shwetabhandare** [comment]
👍

**Pull request merged**